.github/workflows/pullrequest.yml

```diff
@@ -0,0 +1,49 @@
+name: Pull Request
+
+on:
+  pull_request:
+    paths-ignore:
+      - '*.md'
+  workflow_dispatch:
+
+concurrency:
+  group: build-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3.5.0
+        with:
+          distribution: 'temurin'
+          java-version: 11
+
+      - name: Setup gradle
+        uses: gradle/gradle-build-action@v2
+
+      - name: Ktlint
+        run: ./gradlew lintKotlin --stacktrace
+
+      - name: Check Lint
+        run: ./gradlew lintDebug --stacktrace
+
+      - name: Build all build types
+        run: ./gradlew assemble --stacktrace
+
+      - name: Run Bucketeer unit tests
+        run: ./gradlew :bucketeer:testDebugUnitTest --stacktrace
```

Comment:
The workflow for pull-request will run lint, assemble, and unit tests. Integration tests will be executed on main branch.